lib/widgets/lightbox.dart

```diff
@@ -89,6 +90,46 @@ class _CopyLinkButton extends StatelessWidget {
   }
 }
 
+Future<XFile> _downloadImage(Uri url, Map<String, String> headers) async {
+  final response = await http.get(url, headers: headers);
+  final bytes = response.bodyBytes;
+  return XFile.fromData(bytes,
+    name: url.pathSegments.last,
```

Comment:
Missed this in last review — trying to get the filename this way seems fragile.

To do this correctly we will need changes in the content parser (`lib/model/content.dart`) to read the filename in `ImageNode`, but it may be out of scope for this PR. So, for now how about not populating the `name` field here, and leaving a TODO comment.